### PR TITLE
Use 32 threads in single node performance benchmark

### PR DESCRIPTION
**Pull request by igor-aptos**
Also support easier recalibration with just pasting the results.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: testsuite/single_node_performance.py [hunk 1]
```diff
@@ -36,13 +36,76 @@ class Flow(Flag):
...
+NOISE_LOWER_LIMIT = 0.98 if IS_MAINNET else 0.8
+NOISE_LOWER_LIMIT_WARN = None if IS_MAINNET else 0.9
+# If you want to calibrate the upper limit for perf improvement, you can
+# increase this value temporarily (i.e. to 1.3) and readjust back after a day or two of runs
+NOISE_UPPER_LIMIT = 5 if IS_MAINNET else 1.15
+NOISE_UPPER_LIMIT_WARN = None if IS_MAINNET else 1.05
+
+# bump after a perf improvement, so you can easily distinguish runs
+# that are on top of this commit
+CODE_PERF_VERSION = "v5"
```
Comment (msmouse) [inline testsuite/single_node_performance.py:67]: 🚀

File: testsuite/single_node_performance.py [hunk 2]
```diff
@@ -55,36 +118,72 @@ class RunGroupConfig:
...
 # numbers are based on the machine spec used by github action
+# Local machine numbers will be different.
+#
 # Calibrate using median value from
-# Axiom: https://app.axiom.co/aptoslabs-hghf/explorer?qid=88fegG0H1si-s3x8pv&relative=1
 # Humio: https://gist.github.com/igor-aptos/7b12ca28de03894cddda8e415f37889e
-# Local machine numbers will be higher.
-# For charts over time, you can modify the following query:
-# https://app.axiom.co/aptoslabs-hghf/explorer?qid=29zYzeVi7FX-s4ukl5&relative=1
+# Exporting as CSV and copying to the table below.
```
Comment (msmouse) [inline testsuite/single_node_performance.py:130]: 👍 👍 👍

**Review by msmouse — APPROVED** [review]
So we basically no longer need to update the numbers on the tests themselves?

**Review by sitalkedia — APPROVED** [review]

**Comment by igor-aptos** [comment]
> So we basically no longer need to update the numbers on the tests themselves?

updated PR to remove numbers in the tests, to avoid confusion

**Comment by msmouse** [comment]
now I'm confused between block_size and expected_tps.. why not just assume expected_tps=0 if not calibrated yet..

**Comment by igor-aptos** [comment]
we use expected_tps for two things - to assert limits, and as a hint for block size.

we use block_size = min (10000, expected_tps), i.e. so that if we have some transactions with TPS of 20, every block takes at most 1s, and not forever. 

but i agree, this is just a bit too confusing , since we already have waived=True. I'll just remove block_size.

File: testsuite/single_node_performance.py [hunk 3]
```diff
@@ -544,9 +554,16 @@ def print_table(
 
         if test.key in calibrated_expected_tps:
             expected_tps = calibrated_expected_tps[test.key]
+            cur_block_size = int(min([expected_tps, MAX_BLOCK_SIZE]))
         else:
             print(f"WARNING: Couldn't find {test.key} in calibration results")
-            expected_tps = test.expected_tps
+            if test.expected_tps is not None:
+                expected_tps = test.block_size
```
Comment (msmouse) [inline testsuite/single_node_performance.py:561]: ?

**Pull request merged**